### PR TITLE
[docs] [schedules-overhaul] - Add 'how-to' guide for schedules + jobs

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR is part of an overhaul effort to clean up the Schedules concept content. Specifically, this PR adds a 'how-to' guide for using schedules and jobs to automate assets.

## How I Tested These Changes

👀

### CODE REVIEW

File: docs/content/concepts/automation/schedules/running-assets.mdx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+defs = Definitions(
+    assets=[orders_asset, users_asset],
+    jobs=[ecommerce_asset_job],
+    schedules=[ecommerce_schedule],
+)
+```
+
+At this point, our code looks like this:
+
+```python file=concepts/partitions_schedules_sensors/schedules/basic_asset_schedule.py lines=3-10,14-23,30-34,41-47,54-60
```
Comment (PedramNavid) [inline docs/content/concepts/automation/schedules/running-assets.mdx:115]: I added a new property `hidecomments` so you can do `hidecomments=true` in my PR. Even if we don't merge my docs changes, I can create a PR for just the `python file...` enhancement so you don't have to specify every single line next time.

File: docs/content/concepts/automation/schedules/running-assets.mdx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    jobs=[ecommerce_asset_job],
+    schedules=[ecommerce_schedule],
+)
+```
+
+---
+
+## Step 4: Turn the schedule on
+
+Schedules need to be turned on before can be used. In our case, we already turned the schedule on by using the `default_status` parameter in its <PyObject object="ScheduleDefinition" />, but there are a few other ways to do this:
```
Comment (PedramNavid) [inline docs/content/concepts/automation/schedules/running-assets.mdx:159]: before **they** can

**Review by PedramNavid — APPROVED** [review]

File: docs/content/concepts/automation/schedules/running-assets.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/concepts/automation/schedules/running-assets.mdx:115]: omg. That would be amazing.

But this also removes the individual imports from the iterative portions of the code... if we could specify removing only specific lines, that would also be wonderful.

File: docs/content/concepts/automation/schedules/running-assets.mdx (hunk 2 above)
Comment (erinkcochran87) [reply docs/content/concepts/automation/schedules/running-assets.mdx:159]: Good catch!

**Comment by erinkcochran87** [comment]
hyperlint review

**Comment by bllchmbrs** [comment]
@erinkcochran87 I can't delete or edit the hyperlint comment above BUT [I fixed the issue](https://docs.hyperlint.com/changelog.html#_2024-02-22).

Now we can kindly ask hyperlint for a review and unless I broke something in the process, `.mdx` support should be good to go.

hyperlint review

File: docs/content/concepts/automation/schedules/running-assets.mdx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+## Prerequisites
+
+To complete the steps in this guide, you'll need:
+
+- **To have Dagster and the Dagster UI installed.** Refer to the [Installation guide](/getting-started/install) for more info and instructions.
+- **Familiarity with**:
+  - [Software-defined Assets](/concepts/assets/software-defined-assets)
+  - [Jobs](/concepts/ops-jobs-graphs/jobs)
+  - [Code locations](/concepts/code-locations) (<PyObject object="Definitions" />)
```
[the remark this answers is not in the document]
Comment (tacastillo) [reply docs/content/concepts/automation/schedules/running-assets.mdx:29]: rip

File: docs/content/concepts/automation/schedules/running-assets.mdx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+   - Is attached to the `ecommerce_assets_job` job
+   - Has a cron expression of `15 5 * * 1-5`, which translates to `Every Monday through Friday of every month at 5:15AM`
+   - Is turned on by default (`default_status`). We'll discuss this more in [Step 4](#step-4-turn-the-schedule-on).
+
+---
+
+## Step 3: Update the Definitions object
+
+Next, we'll update our project's <PyObject object="Definitions" /> object to include our new job and schedule. This ensures the job and schedule are available to Dagster processes like the Dagster UI.
```
[the remark this answers is not in the document]
Comment (cmpadden) [reply docs/content/concepts/automation/schedules/running-assets.mdx:100]: Looks like `PyObject` components are removed in some places, but not all. 🤔

[the remark this answers is not in the document]
Comment (bllchmbrs) [reply docs/content/concepts/automation/schedules/running-assets.mdx:100]: Yeah, I looked at the logs and at certain places it thinks it's placeholder text while it others it gets that it's part of MIDX. I've got some tuning to do to improve MDX support that I'm confident should help out.

File: docs/content/concepts/automation/schedules/running-assets.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bllchmbrs) [reply docs/content/concepts/automation/schedules/running-assets.mdx:29]: https://github.com/dagster-io/dagster/pull/19175#discussion_r1500839089

Working on better MDX support now.

**Comment by erinkcochran87** [comment]
hyperlint review

**Comment by bllchmbrs** [comment]
@erinkcochran87 , let me know if you want to override this!

Here's the new review structure: https://github.com/Hyperlint-AI/demo/pull/4
@cmpadden , would love your feedback as well!

**Comment by erinkcochran87** [comment]
Closing in favor of #20346, so I don't have to untangle outdated branches

**Pull request closed without merging**